mogp_emulator/Priors.py

```diff
@@ -118,44 +218,56 @@ def corr(self, newcorr):
             if not issubclass(type(d), WeakPrior):
                 raise TypeError("Correlation priors must be a list of WeakPrior derived objects")
         self._corr = list(newcorr)
+        if not self.n_corr is None and not self.n_corr == len(self._corr):
+            print("Length of corr argument differs from specified value of n_corr. " +
+                  "Defaulting to the value given by the corr argument.") 
         self._n_corr = len(self._corr)
         
     @property
     def n_corr(self):
+        """
+        Number of correlation length parameters
+        """
         return self._n_corr
     
     @property
     def cov(self):
         """Covariance Scale Priors
         
-        Prior distribution on Covariance Scale.
+        Prior distribution on Covariance Scale. Can be set using a ``WeakPriors``
+        derived object.
         """
         return self._cov
         
     @cov.setter
     def cov(self, newcov):
+        "Setter method for cov"
         if newcov is None:
             newcov = WeakPrior()
         if not issubclass(type(newcov), WeakPrior):
             raise TypeError("Covariance prior must be a WeakPrior derived object")
         self._cov = newcov
-    
-    @property
-    def fit_cov(self):
-        "Property indicating if the covariance must be fit or can be found analytically"
-        return True
-        #return (not (self.mean.has_weak_priors and (type(self.cov) is WeakPrior or type(self.cov) is InvGammaPrior)))
-    
+
     @property
     def nugget_type(self):
+        """
+        Nugget fitting method for the parent GP.
+        """
         return self._nugget_type
     
     @property
     def nugget(self):
+        """
+        Nugget prior distribution
+        
+        If a nugget is fit, this determines the prior used. If the nugget
+        is not fit, will automatically set this to ``None``.
+        """
         return self._nugget
         
     @nugget.setter
     def nugget(self, newnugget):
+        "Setter method for nugget"
```

Comment:
Yes, probably a good idea. I may have been somewhat lax with some of these docstrings because of how autodoc handles attributes with a property decorator -- it just inserts the docstring of the getter method so you have to put all of the information in that one. However, you are right that having something more specific inline in the code is helpful.